### PR TITLE
Upgradeable IpAssetRegistry

### PR DESCRIPTION
**Pull request by Ramarti**
Refactor to make IPAssetRegistry upgradeable

### CODE REVIEW

File: contracts/registries/IPAssetRegistry.sol [hunk 1]
```diff
@@ -21,16 +24,32 @@ import { IPAccountStorageOps } from "../lib/IPAccountStorageOps.sol";
...
+    struct IPAssetRegistryStorage {
+        uint256 totalSupply;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("story-protocol.IPAssetRegistry")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 private constant IPAssetRegistryStorageLocation =
+        0x987c61809af5a42943abd137c7acff8426aab6f7a1f5c967a03d1d718ba5cf00;
 
+    /// @custom:oz-upgrades-unsafe-allow constructor
     constructor(address erc6551Registry, address ipAccountImpl) IPAccountRegistry(erc6551Registry, ipAccountImpl) {}
```
Comment (kingster-will) [inline contracts/registries/IPAssetRegistry.sol:44]: Do we need `_disableInitializers` in this constructor?

File: contracts/registries/IPAssetRegistry.sol [hunk 2]
```diff
@@ -21,16 +24,32 @@ import { IPAccountStorageOps } from "../lib/IPAccountStorageOps.sol";
...
 
+    /// @notice Initializes the IPAssetRegistry contract.
+    /// @param accessManager The address of the access manager.
+    function initialize(address accessManager) public initializer {
```
Comment (LeoHChen) [inline contracts/registries/IPAssetRegistry.sol:48]: I saw there was a check if accessManager = address(0) in other PR.  Do you need to add this check here?
https://github.com/storyprotocol/protocol-core-v1/pull/60/files#diff-cd57b00a1247e3ce32fcf5e25ba6a371cecbee670026706be6563edca4bbec4aR96

**Comment by jdubpark** [comment]
closes #49

File: contracts/registries/IPAssetRegistry.sol (hunk 1 above)
Comment (jdubpark) [reply contracts/registries/IPAssetRegistry.sol:44]: Need it

Comment (Ramarti) [reply contracts/registries/IPAssetRegistry.sol]: b020807d33f4dfe9c1fa5f14a0567f07624a2d9e

File: contracts/registries/IPAssetRegistry.sol (hunk 2 above)
Comment (Ramarti) [reply contracts/registries/IPAssetRegistry.sol:50]: Fixed b020807d33f4dfe9c1fa5f14a0567f07624a2d9e

**Review by jdubpark — APPROVED** [review]
nit: some zero address reverts are placed after some upgrade init (just code uniformity)

**Review by LeoHChen — APPROVED** [review]

**Pull request merged**